File: ecommerce_client/src/services/bannerService.ts
import api from '@/lib/api'
import type { Banner } from '@/types/banner.types'

export const bannerService = {
  getActiveBanners: async (): Promise<Banner[]> => {
    const response = await api.get<Banner[]>('/banners')
    return response.data
  },

  getAllBanners: async (): Promise<Banner[]> => {
    const response = await api.get<Banner[]>('/banners/all')
    return response.data
  },

  getBannerByCode: async (code: number): Promise<Banner> => {
    const response = await api.get<Banner>(`/banners/${code}`)
    return response.data
  },

  createBanner: async (banner: Partial<Banner>): Promise<Banner> => {
    const response = await api.post<Banner>('/banners/save', banner)
    return response.data
  },

  updateBanner: async (banner: Partial<Banner>): Promise<Banner> => {
    const response = await api.post<Banner>('/banners/update', banner)
    return response.data
  },

  deleteBanner: async (code: number): Promise<boolean> => {
    const response = await api.delete<boolean>(`/banners/remove/${code}`)
    return response.data
  }
}

File: microservice/product-service/src/main/java/com/shop/product_service/banner/service/BannerService.java
package com.shop.product_service.banner.service;

import com.shop.product_service.banner.dto.BannerDto;
import com.shop.product_service.banner.dto.response.BannerDtoResponse;

import java.util.List;

public interface BannerService {

    List<BannerDtoResponse> getActiveBanners();

    List<BannerDtoResponse> getAllBanners();

    BannerDtoResponse getBannerByCode(Long code);

    BannerDtoResponse createBanner(BannerDto bannerDto);

    BannerDtoResponse updateBanner(BannerDto bannerDto);

    boolean deleteBanner(Long code);
}
